src/App.js

```diff
@@ -109,6 +116,18 @@ class App extends Component {
     });
   }
 
+  updateInitiative(initiative, player) {
+    console.log('Set ' + player.title + ' initiative to ' + initiative);
+    player.initiative = initiative;
+    const playerStateIndex = this.state.players.find(
+      item => item.title === player.title,
+    );
+
+    this.setState({
+      playerStateIndex: player,
```

Comment:
Instead of modifying the `player` argument (which is passed by reference, and so might end up modifying something you're not expecting it to elsewhere), use destructuring here to create a new object with the properties of `player` and the new initiative:

```js
this.setState({
    playerStateIndex: {
        ...this.state[playerStateIndex],
        initiative: initiative,
    },
});
```

Reply:
You're right, the player reference initiative being changed was actually what was holding together the updating of the fields in List. hmmmmm.